core/src/main/scala/org/apache/spark/sql/catalyst/planner/TiAggregationImpl.scala

```diff
@@ -84,15 +84,11 @@ object TiAggregationImpl {
               a.copy(aggregateFunction = Count(r.child), resultId = newExprId))
         }.toMap
 
+        // ResultDataType of sum only can be double or decimal
+        // We need to cast decimal back to long for the promotedSum, or spark will regard decimal as long cause the wrong answer
         val sumRewrite = sumsRewriteMap.map {
           case (ref, sum) =>
-            val resultDataType = sum.resultAttribute.dataType
-            val castedSum = resultDataType match {
-              case LongType => Cast(sum.resultAttribute, DecimalType.BigIntDecimal)
-              case FloatType => Cast(sum.resultAttribute, DoubleType)
-              case DoubleType => Cast(sum.resultAttribute, DoubleType)
-              case d: DecimalType => Cast(sum.resultAttribute, d)
-            }
+            val castedSum = Cast(sum.resultAttribute, ref.dataType)
```

Comment:
Will it throw exception when value is overflow

Reply:
we can't control the behavior when overflow, it depends on the logic of spark cast operator.

Reply:
And you can find the error results also happens in Spark JDBC rather than exception when overflow